src/morphology.rs

```diff
@@ -430,12 +430,25 @@ impl Mask {
             image.height() < 512,
             "the input image must be at most 511 pixels high"
         );
-        let (center_x, center_y) = (i16::from(center_x), i16::from(center_y));
+        let center = Point::new(center_x, center_y).to_i16();
         let elements = image
             .enumerate_pixels()
             .filter(|(_, _, &p)| p[0] != 0)
-            .map(|(x, y, _)| Point::new(x as i16 - center_x, y as i16 - center_y)) // cast to i16 ok because of the size check
+            .map(|(x, y, _)| Point::new(x, y).to_i16())
+            .map(|p| p - center)
             .collect();
+        Self::new(elements)
+    }
+
+    fn new(elements: Vec<Point<i16>>) -> Self {
+        assert!(elements.len() <= (511 * 511) as usize);
+        debug_assert!({
+            let unique = elements
+                .iter()
+                .map(|p| (p.x, p.y))
+                .collect::<std::collections::HashSet<_>>();
+            unique.len() == elements.len()
+        });
```

Comment:
```suggestion
        debug_assert!(elements.iter().all_unique());
```
Use `Itertools` to make this more readable.